### ISSUE TITLE
Installer in FFS 11.6

### ISSUE THREAD

**Issue by Garbulix** [flathub/org.freefilesync.FreeFileSync#46]
Something happened to FFS for Linux - now its distribued by binary(?) .run installer. I remember you had many problems when compiling program, thus I am almost sure I wouldn't be able to help with this release. 
Is there another way with given files from the publisher?

**Comment by kparal** [comment]
![facepalm](https://i.stack.imgur.com/jiFfM.jpg)

**Comment by Garbulix** [comment]
One facepalm for me and one for Zenju? :D

**Comment by Garbulix** [comment]
I think creating Flatpak releases for FFS looks a bit like "battle"(?) - I dont have better words. 
I think it is because the developer wants to get money from his Donate edition, which has built-in updater.

**Comment by kparal** [comment]
> One facepalm for me and one for Zenju? :D

Both for the developer. I'm getting really tired with the nonsense he (or she) constantly produces. If I knew a good alternative to FFS, I'd have dropped packaging it a long time ago.

I'll have a look at the new package if we can do something about it.

**Comment by kparal** [comment]
Not only that it's an installer, it also requires `root` to be installed:
```
$ ./FreeFileSync_11.6_Install.run
Please log in as superuser in order to install FreeFileSync:
[sudo] password for kparal: 
```
Not even specifying a user directory avoids that:
```
$ ./FreeFileSync_11.6_Install.run --help
FreeFileSync installation parameters:
    -h, --help    Show help
    --directory   Change installation directory e.g. --directory /opt/FFS
    --noshortcuts  Don't create desktop shortcuts

$ ./FreeFileSync_11.6_Install.run --directory /home/kparal/tmp/FFS
Please log in as superuser in order to install FreeFileSync:
[sudo] password for kparal: 
```

There's only one way to respond to that:
![triple facepalm](https://i.pinimg.com/originals/53/11/b4/5311b45db9177059bab9ae8a32aa063c.jpg)

**Comment by kparal** [comment]
I just tested, flatpak builds the package as a regular user, not root. The installer can't be unpacked during the build process:
```
========================================================================
Building module freefilesync in /home/kparal/devel/flatpak/org.freefilesync.FreeFileSync/.flatpak-builder/build/freefilesync-3
========================================================================
Running: ./FreeFileSync_11.6_Install.run --directory /app/FreeFileSync
Please install "sudo" or run this installer as root.

/tmp/FreeFileSync_11.6_ncOAnF/install.sh: line 75: sudo: command not found
```

There were multiple reasons why I switched from building from source to packaging pre-built binaries, including:
a) no source version control available
b) no proper build system (like cmake or meson), just a single makefile, which often required heavy patching to work
c) the developer modifies system libraries like curl, ssh or wxwidgets, but doesn't supply their source, just an unparseable diff described in `Bugs.txt` in the tarball. That either causes the build to fail, or makes the runtime behavior of FFS different from the upstream version. Some developers (e.g. [Arch ones](https://aur.archlinux.org/packages/freefilesync/)) even go as far as to patch FFS source code to not rely on the patched library functionality (again, at the expense of a different runtime behavior). Projects should never depend on privately patched libraries (and if they do, at least bundle them), yet FFS does exactly that. It's sad to read Arch devs comments like "_After many nights of plowing through code i managed to get the horror release 11.1 done_".
d) it's clear that the source tarball is actually not used to build the binaries, because sometimes it outright fails to build because of errors and the developer sometimes corrects that afterwards. But the binaries must have been built from a different source in that case. The source code dump is apparently an afterthought for the developer, and incorrect practices are used throughout the whole process.

For this reason I don't intend to go back to building from source. I don't have time nor nerves for that crap. Of course if somebody is willing to step in a handle all of this, I'll be more than happy.

I suggest we wait some time whether the developer retracts on his/her decision to provide a root-only installer for Linux (and no binary tarball or at least a user-only installer). If that doesn't happen, I'd say we should remove FFS from Flathub and start looking for an alternative tool (unfortunately, because I really like FFS).

**Comment by Garbulix** [comment]
I don't think anyone from FFS forum will have problem with the installer. 
I've sent author an email about new installer and if hshe would distribute program as portable compiled app (like in previous releases), but I am not sure if I would get the reply. I wanted to create forum topic, but bot assumed me as spammer.

For now, I am supposing to find an alternative (I guess "forking" and distribuing the app as a new project would be too difficult), because Zenju doesnt seem to be that community-aimed (but in fact I am not sure if there were enough people to talk about it with shim).

**Comment by seth100** [comment]
Hi, I saw the new installer and I'm gonna start doing something that I was planning to do time ago which is STOP using this tool! 
It will be a very hard process for me especially because I have many configuration profiles and I was used to this tool, but I hate that I have to install it with root privileges, that the flatpak installer has now issues in updating to new version and besides it has a [huge issue](https://github.com/flathub/org.freefilesync.FreeFileSync/issues/35) (even if not related to FreeFileSync itself AFAIK).
Do you know any good alternative?

**Comment by Garbulix** [comment]
Currently I try to use **unison-gtk** (available in [some repositories](https://pkgs.org/search/?q=unison)). It is not that intuitive, but it is suitable for my simple mirror syncs. There also is **Grsync**, which will be tested I hope.
But I am afraid there is no good alternative.
My current plan is to wait for some kind of reply from Zenju - maybe [on forum](https://freefilesync.org/forum/viewtopic.php?t=8101) (which is very unlikely IMO). Or maybe the author will be distribuing tarball as in previous versions starting from version 11.7.
But even though there will be a tarball available again, we will have a license problem, because [I saw in FAQ](https://freefilesync.org/faq.php#distribute) that the program author doesn't want to distribue it by some kind of auto-updated repositories and, again IMO, it is very unlikely that Zenju will allow us to distribute FFS by Flatpak.

Anyway - for now we wait. It is also a good idea to look for some kind of suitable alternative in the meantime.

**Comment by seth100** [comment]
yes let's see what he says, in the meanwhile it seems that Grsync is exactly what I need, thank you @Garbulix for mentioning that!

**Comment by Garbulix** [comment]
As bgstack15 said on forum [HERE](https://freefilesync.org/forum/viewtopic.php?t=8135), installer can be extracted with 7zip.  
`7z x -oFFS_116 ./ffs_installer.run`
Where `ffs_installer.run` is installer provided by Zenju. File will be unpacked to folder `FFS_116`, and there will be a tar.gz file, which seems to be a portable tarball like in previous versions.

**Comment by kparal** [comment]
> `7z x -oFFS_116 ./ffs_installer.run`

Interesting, I can confirm this. Thanks for the update.

Before we start updating the flatpak package to 11.6, we need to resolve the license issue, though. I created a separate ticket for it, see #48 .

**Comment by kparal** [comment]
Since the pre-compiled binary distribution permission was denied (see #48), I'm closing this issue.

**Comment by kparal** [comment]
We've identified a solution for the license problem in #48, and so I'm reopening this issue.
